File: Models/Discount.cs
namespace web_agent_pro.Models
{
    public partial class Discount
    {
        public long DiscountId { get; set; }
        public string Name { get; set; }
        public string Scope { get; set; }
        public double Amount { get; set; }
        public string State { get; set; }
    }
}

File: ClientApp/src/data/models/domain/discount.ts
export class Discount {
  constructor(name: string, scope: string, amount: number, state: string) {
    this.name = name;
    this.scope = scope;
    this.amount = amount;
    this.state = state;
  }

  discountId: number;
  name: string;
  scope: string;
  amount: number;
  state: string;
}
